tests/subtle/reaction.test.ts

```diff
@@ -0,0 +1,142 @@
+import { describe, test, assert } from "vitest";
+import { Signal } from "signal-polyfill";
+import { reaction } from "../../src/subtle/reaction.ts";
+
+describe("reaction()", () => {
+
+  test("calls the effect function when the data function return value changes", async () => {
+    const count = new Signal.State(0);
+
+    let callCount = 0;
+    let value, previousValue;
+
+    reaction(() => count.get(), (_value, _previousValue) => {
+      callCount++;
+      value = _value;
+      previousValue = _previousValue;
+    });
+
+    // Effect callbacks are not called immediately
+    assert.strictEqual(callCount, 0);
+
+    await 0;
```

Comment:
what's this do? could a comment be added?

alternatively, there is a more semantically appropriate test utility already:
https://github.com/proposal-signals/signal-utils/blob/main/tests/helpers.ts#L79

```js
await waitForMicrotask()`
```

Reply:
Awaiting anything waits for at least a microtask. This is just a no-dependency way of doing that.

That utility does more than it needs to. It could just use `Promise.resolve()`:

```ts
export function waitForMicrotask() {
  return Promise.resolve();
}
```

but it could also be an empty function:

```ts
export async function waitForMicrotask() {}
```

Reply:
Added a comment